runtime/src/iree/hal/buffer_heap.c

```diff
@@ -185,13 +185,13 @@ iree_status_t iree_hal_heap_buffer_wrap(
   IREE_ASSERT_ARGUMENT(out_buffer);
   IREE_TRACE_ZONE_BEGIN(z0);
 
-  if (!iree_host_size_has_alignment((uintptr_t)data.data,
+  if (!iree_host_size_has_alignment((uintptr_t)data.data_length,
```

Comment:
this is incorrect - the base of the buffer pointer is what needs to be aligned, not the size
if the pointers coming back from wgpuBufferGetConstMappedRange are unaligned that's likely to be a problem (but one we could work around in the short term with a flag that disables this check)

Reply:
Ahh, I see... yeah, the pointers are aligned to 8, not 64:
```
buffer_heap.c:197: OUT_OF_RANGE; imported heap buffer data must be aligned to 64; got 0x52d008
```

Could I relax the alignment requirement for heap buffers on the Emscripten platform (or just for this sample)? https://github.com/iree-org/iree/blob/a0c6bd575cb011468f5b4aaaa1339ce944bfc558/runtime/src/iree/base/config.h#L174-L180

Reply:
I'm wondering if there's something we can pass to `requiredLimits` (https://gpuweb.github.io/gpuweb/#dictdef-gpudevicedescriptor).

https://gpuweb.github.io/gpuweb/#limits only mentions these though:
* `minUniformBufferOffsetAlignment` (default 256 bytes)
* `minStorageBufferOffsetAlignment` (default 256 bytes)

Reply:
Oh, we might be okay after the other changes... I don't see that error any longer.

Reply:
Oh wait, I don't see the error for fullyconnected (5x1x5xf32 output), but I still see it for simple_abs (f32 output)
`buffer_heap.c:194: OUT_OF_RANGE; imported heap buffer data must be aligned to 64; got 0x52cf10` (aligned to 8 or 16)